contracts/src/libraries/token/erc677.cairo

```diff
@@ -30,37 +37,42 @@ mod ERC677 {
     #[event]
     #[derive(Drop, starknet::Event)]
     enum Event {
-        Transfer: Transfer,
+        TransferAndCall: TransferAndCall,
     }
 
     #[derive(Drop, starknet::Event)]
-    struct Transfer {
+    struct TransferAndCall {
         from: ContractAddress,
         to: ContractAddress,
         value: u256,
         data: Array<felt252>
     }
 
-    fn transfer_and_call(
-        ref self: ContractState, to: ContractAddress, value: u256, data: Array<felt252>
-    ) -> bool {
-        let sender = starknet::info::get_caller_address();
+    #[embeddable_as(ERC677Impl)]
+    impl ERC677<
+        TContractState, +HasComponent<TContractState>, +IERC20<TContractState>, +Drop<TContractState>,
```

Comment:
neat how you can enforce that the contract must implement ERC20 Interface. 

But why does ContractState need to be droppable? It's always used through a mutable reference or snapshot right?

Reply:
I'm not quite sure why but it's required to get the code compiling. There's some examples here https://github.com/ArielElp/component-examples